From fdd8621672eadecde31b76b6bc52ca6a8d745718 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 19 Dec 2012 12:43:19 +0900
Subject: [PATCH 96/97] ext/dfbvideosink: don't use the cursor in the
 exclusive mode

When using the exclusive mode, EnableCusor() wouldn't be available
in the DirectFB 1.6.0 and later.
This patch determines to invoke EnableCusor() if the layer mode is
the administrative.
---
 ext/directfb/dfbvideosink.c |   10 ++++++++--
 ext/directfb/dfbvideosink.h |    4 ++++
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index e7589a4..39aa4cc 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -886,7 +886,10 @@ gst_dfbvideosink_setup (GstDfbVideoSink * dfbvideosink)
       dfbvideosink->layer->SetBackgroundColor (dfbvideosink->layer,
           0x00, 0x00, 0x00, 0xFF);
 
-      dfbvideosink->layer->EnableCursor (dfbvideosink->layer, TRUE);
+#if (DIRECTFB_VER >= GST_DFBVIDEOSINK_VER (1,6,0))
+      if (dfbvideosink->layer_mode == LAYER_MODE_ADMINISTRATIVE)
+#endif
+        dfbvideosink->layer->EnableCursor (dfbvideosink->layer, TRUE);
 
       /* We prefer a double buffered primary surface */
       if (dfbvideosink->backbuffer) {
@@ -1091,7 +1094,10 @@ gst_dfbvideosink_cleanup (GstDfbVideoSink * dfbvideosink)
   }
 
   if (dfbvideosink->layer) {
-    dfbvideosink->layer->EnableCursor (dfbvideosink->layer, FALSE);
+#if (DIRECTFB_VER >= GST_DFBVIDEOSINK_VER (1,6,0))
+    if (dfbvideosink->layer_mode == LAYER_MODE_ADMINISTRATIVE)
+#endif
+      dfbvideosink->layer->EnableCursor (dfbvideosink->layer, FALSE);
     dfbvideosink->layer->Release (dfbvideosink->layer);
     dfbvideosink->layer = NULL;
   }
diff --git a/ext/directfb/dfbvideosink.h b/ext/directfb/dfbvideosink.h
index 2284002..46f2308 100644
--- a/ext/directfb/dfbvideosink.h
+++ b/ext/directfb/dfbvideosink.h
@@ -23,6 +23,10 @@
 #include <gst/video/gstvideosink.h>
 
 #include <directfb.h>
+#include <directfb_version.h>
+
+#define GST_DFBVIDEOSINK_VER(a,b,c) (((a) << 16) | ((b) << 8) | (c))
+#define DIRECTFB_VER GST_DFBVIDEOSINK_VER(DIRECTFB_MAJOR_VERSION,DIRECTFB_MINOR_VERSION,DIRECTFB_MICRO_VERSION)
 
 #define LAYER_MODE_INVALID          -1
 #define LAYER_MODE_EXCLUSIVE        DLSCL_EXCLUSIVE
-- 
1.7.10.4

